packages/commonwealth/server/util/verifySessionSignature.ts

```diff
@@ -1,346 +1,51 @@
-    log.error('no community provided to verifySignature');
-    return false;
-  }
-
-  // Reconstruct the expected canvas message.
-  const canvasChainId = chainBaseToCanvasChainId(community.base, chain_id);
-
-  const canvasSessionPayload = createCanvasSessionPayload(
-    community.base,
-    canvasChainId,
-    community.base === ChainBase.Substrate
-      ? addressSwapper({
-          address: addressModel.address,
-          currentPrefix: 42,
-        })
-      : addressModel.address,
-    // @ts-expect-error StrictNullChecks
-    sessionAddress,
-    // @ts-expect-error StrictNullChecks
-    parseInt(sessionIssued, 10),
-    sessionBlockInfo
-      ? addressModel.block_info
-        ? JSON.parse(addressModel.block_info).hash
-        : null
-      : null,
-  );
-
-  let isValid: boolean;
-  if (community.base === ChainBase.Substrate) {
-    //
-    // substrate address handling
-    //
-
-    // Import @polkadot/keyring twice, since the first import might fail invisibly
-    // because of conflicting package versions, causing signerKeyring.verify to fail.
-    await import('@polkadot/keyring');
-    const polkadot = await import('@polkadot/keyring');
-    const address = polkadot.decodeAddress(addressModel.address);
-    const keyringOptions: KeyringOptions = { type: 'sr25519' };
-    if (
-      !addressModel.keytype ||
-      addressModel.keytype === 'sr25519' ||
-      addressModel.keytype === 'ed25519'
-    ) {
-      if (addressModel.keytype) {
-        keyringOptions.type = addressModel.keytype as KeypairType;
-      }
-      keyringOptions.ss58Format = community.ss58_prefix ?? 42;
-      const signerKeyring = new polkadot.Keyring(keyringOptions).addFromAddress(
-        address,
-      );
-      const message = stringToHex(sortedStringify(canvasSessionPayload));
+  const signers = getSessionSigners();
 
-      const signatureU8a =
-        signatureString.slice(0, 2) === '0x'
-          ? hexToU8a(signatureString)
-          : hexToU8a(`0x${signatureString}`);
-      isValid = signerKeyring.verify(message, signatureU8a, address);
-    } else {
-      log.error('Invalid keytype.');
-      isValid = false;
-    }
-  } else if (
-    community.base === ChainBase.CosmosSDK &&
-    (addressModel.wallet_id === WalletId.CosmosEvmMetamask ||
-      addressModel.wallet_id === WalletId.KeplrEthereum)
-  ) {
-    //
-    // ethereum address handling on cosmos chains via metamask
-    //
-    const msgBuffer = Buffer.from(sortedStringify(canvasSessionPayload));
-
-    // toBuffer() doesn't work if there is a newline
-    const msgHash = ethUtil.hashPersonalMessage(msgBuffer);
-    const ethSignatureParams = ethUtil.fromRpcSig(signatureString.trim());
-    const publicKey = ethUtil.ecrecover(
-      msgHash,
-      ethSignatureParams.v,
-      ethSignatureParams.r,
-      ethSignatureParams.s,
+  const expectedAddress = addressModel.address;
+  const sessionAddress = session.address.split(':')[2];
+  if (sessionAddress !== expectedAddress) {
```

Comment:
I think we should add a generic address comparison function that handles the Substrate addressSwapper-involved address comparisions, in lib/utils, once #8020 is merged.

That should probably be in a different PR since it involves code that this PR doesn't touch.

(This change here is just a bugfix since we weren't canonicalizing Polkadot addresses properly in the revalidation modal.)

Reply:
sounds good